File: 1701-average-waiting-time/1701-average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double waiting = 0;
        int last_finish = customers[0][0];
        for (int i = 0; i < customers.size(); i++){
            int extra_time = 0;
            int arrival = customers[i][0];
            int time = customers[i][1];

            if (last_finish <= arrival){
                waiting += time;
                last_finish = arrival + time;
            }
            else{
                extra_time = last_finish - arrival;
                waiting += time + extra_time;
                last_finish += time;
            }
        }
        return waiting / customers.size();    
    }
};

File: 1701-average-waiting-time/1701-average-waiting-time.py
from typing import List
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        waiting = 0 
        last_finish = customers[0][0]
        for arrival, time in customers:
            extra_time = 0
            if last_finish <= arrival:
                waiting += time
            else:
                extra_time = last_finish - arrival
                waiting += time + extra_time

            if last_finish < arrival:
                last_finish = arrival + time
            else:
                last_finish += time
        return waiting / len(customers)

